src/blocks/store/blob.clj

```diff
@@ -36,27 +38,101 @@
   [root id]
   (str root (multihash/hex id)))
 
-(defn blob->stats
+
+(defn- blob->stats
+  "Return the stats map (with metadata) for a blob."
   [^CloudBlob blob]
-  (let [^BlobProperties properties (.getProperties blob)
-        name (last (str/split (.getName blob) #"/"))]
-    {:id (multihash/decode name)
-     :size (.getLength properties)
-     :source (.getPrimaryUri (.getSnapshotQualifiedStorageUri blob))
-     :stored-at (.getLastModified properties)}))
-
-(defn file->block
-  [^CloudBlobContainer container root stats]
-  (let [id (:id stats)
-        path (id->path root id)]
-    (block/with-stats
-      (data/lazy-block
-        id (:size stats)
-        (fn file-reader
-          []
-          (-> (.getBlockBlobReference container path)
-              (.openInputStream))))
-      (dissoc stats :id :size))))
+  (let [properties (.getProperties blob)
+        blob-name (last (str/split (.getName blob) #"/"))]
+    (when (hex? blob-name)
+      (with-meta
+        {:id (multihash/parse blob-name)
+         :size (.getLength properties)
+         :stored-at (if-let [date (or (.getLastModified properties)
+                                      (.getCreatedTime properties))]
+                      (.toInstant ^Date date)
+                      (Instant/now))}
+        {::source (.getPrimaryUri (.getSnapshotQualifiedStorageUri blob))
+         ::name (.getName blob)}))))
+
+
+
+;; ## Blob Content
+
+(deftype BlobReader
+  [^CloudBlockBlob blob]
+
+  data/ContentReader
+
+  (read-all
+    [this]
+    (log/trace "Opening Azure blob" (.getName blob))
+    #_(.getBlockBlobReference container blob-name)
```

Comment:
Should this be committed?

Reply:
Ah, good spot - this was left over from checking whether the `CloudBlob` object was sufficient to wrap or if we needed to re-fetch it from the container before each access.